### ISSUE TITLE
OOMKilled with sync of 100GB file, RC=137

### ISSUE THREAD

**Issue by akmithal** [noobaa/noobaa-core#6731]
### Environment info
- NooBaa Version: master-20210921
- Platform: OCP 4.8.9

### Actual behavior
1. s3 sync behaviour caused OOMKilled

### Expected behavior
1. No IO failure or OOM error for endpoint pod 

### Steps to reproduce

1. Started sync operation from s3://bucket-1 to s3://bucket-21 and to s3://bucket-11/ for 100GB file 

`copy failed: s3://bucket-1/big_file_100GB to s3://bucket-21/big_file_100GB An error occurred (503) when calling the UploadPartCopy operation: Service Unavailable
`

2. Got endpoint OOMError in few minutes
```
[root@ocp-akshat-1-inf ~]# podn
NAME                                               READY   STATUS        RESTARTS   AGE
noobaa-core-0                                      1/1     Running       0          24h
noobaa-db-pg-0                                     1/1     Running       0          25h
noobaa-default-backing-store-noobaa-pod-173508df   0/1     Terminating   0          16h
noobaa-endpoint-c79f5477c-wmdvh                    0/1     OOMKilled     1          24h
noobaa-operator-5d7dc449f8-gtr7s                   1/1     Running       8          25h

```

Endpoint resources are 2CPU, 3 GB RAM. 

From endpoint log:

```
Sep-23 7:02:53.966 [Endpoint/13]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Sep-23 7:02:53.966 [Endpoint/13]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-1 this.bucket_path: /nsfs/nsfs-nsr-1/bucket-11 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Sep-23 7:02:53.966 [Endpoint/13]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Sep-23 7:02:53.966 [Endpoint/13]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-1 this.bucket_path: /nsfs/nsfs-nsr-1/bucket-11 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Sep-23 7:02:54.022 [Endpoint/13]    [L0] core.sdk.object_sdk:: validate_non_nsfs_bucket:  { name: SENSITIVE-7fa8f881394f8a6d, email: SENSITIVE-3d288e97f8a66631, access_keys: [ { access_key: SENSITIVE-c42d7183635a2fdd, secret_key: SENSITIVE-1a0bca9e384192a9 } ], has_login: false, has_s3_access: true, allowed_buckets: { full_permission: true }, default_resource: 'nsfs-nsr-1', can_create_buckets: true, nsfs_account_config: { gid: 800, uid: 5000, nsfs_only: true, new_buckets_path: '/', backend: 'GPFS', warn_threshold_ms: 100 }, systems: [ { name: 'noobaa', roles: [ 'admin' ] } ], external_connections: { count: 0, connections: [] }, preferences: { ui_theme: 'DARK' } } { fs_root_path: '/nsfs/nsfs-nsr-1', fs_backend: 'GPFS', id: '614ac781cc75f3002eb2d295', name: 'nsfs-nsr-1' }
Sep-23 7:02:54.022 [Endpoint/13]    [L1] core.endpoint.s3.s3_rest:: S3 REQUEST PUT /bucket-11/big_file_100GB?uploadId=027ef277-2da0-4fdf-b98b-07d47af90cf4&partNumber=987 op put_object_uploadId request_id ktwl8w1u-1o62qv-pip { 'accept-encoding': 'identity', 'x-amz-copy-source': 'bucket-1/big_file_100GB', 'x-amz-copy-source-range': 'bytes=16542334976-16559112191', 'user-agent': 'aws-cli/2.1.29 Python/3.8.8 Linux/4.18.0-305.17.1.el8_4.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.sync', 'x-amz-date': '20210923T070253Z', 'x-amz-content-sha256': 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', authorization: 'AWS4-HMAC-SHA256 Credential=NR5YaXYtWaETVrQar7Nr/20210923/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-range;x-amz-date, Signature=4228e8b3095d70787cf8cbe3768e316627993f59c0f25009f2e90e848ce6ea0e', 'content-length': '0', host: 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-host': 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-port': '443', 'x-forwarded-proto': 'https', forwarded: 'for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https', 'x-forwarded-for': '10.17.66.193' }
Sep-23 7:02:54.022 [Endpoint/13]    [L0] core.endpoint.s3.ops.s3_put_object_uploadId:: PUT OBJECT PART bucket-11 big_file_100GB 987 bucket-1/big_file_100GB
Sep-23 7:02:54.024 [Endpoint/13]    [L0] core.sdk.object_sdk:: validate_non_nsfs_bucket:  { name: SENSITIVE-7fa8f881394f8a6d, email: SENSITIVE-3d288e97f8a66631, access_keys: [ { access_key: SENSITIVE-c42d7183635a2fdd, secret_key: SENSITIVE-1a0bca9e384192a9 } ], has_login: false, has_s3_access: true, allowed_buckets: { full_permission: true }, default_resource: 'nsfs-nsr-1', can_create_buckets: true, nsfs_account_config: { gid: 800, uid: 5000, nsfs_only: true, new_buckets_path: '/', backend: 'GPFS', warn_threshold_ms: 100 }, systems: [ { name: 'noobaa', roles: [ 'admin' ] } ], external_connections: { count: 0, connections: [] }, preferences: { ui_theme: 'DARK' } } { fs_root_path: '/nsfs/nsfs-nsr-1', fs_backend: 'GPFS', id: '614ac781cc75f3002eb2d295', name: 'nsfs-nsr-1' }
Sep-23 7:02:54.024 [Endpoint/13]    [L1] core.endpoint.s3.s3_rest:: S3 REQUEST PUT /bucket-11/big_file_100GB?uploadId=027ef277-2da0-4fdf-b98b-07d47af90cf4&partNumber=988 op put_object_uploadId request_id ktwl8w1y-1qxqzz-kky { 'accept-encoding': 'identity', 'x-amz-copy-source': 'bucket-1/big_file_100GB', 'x-amz-copy-source-range': 'bytes=16559112192-16575889407', 'user-agent': 'aws-cli/2.1.29 Python/3.8.8 Linux/4.18.0-305.17.1.el8_4.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.sync', 'x-amz-date': '20210923T070253Z', 'x-amz-content-sha256': 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', authorization: 'AWS4-HMAC-SHA256 Credential=NR5YaXYtWaETVrQar7Nr/20210923/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-range;x-amz-date, Signature=c8ed4b7e034d72cfb868052926f8b1a4afd7ce42ce2a2ef80aa2eb01d8969e2f', 'content-length': '0', host: 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-host': 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com', 'x-forwarded-port': '443', 'x-forwarded-proto': 'https', forwarded: 'for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https', 'x-forwarded-for': '10.17.66.193' }
Sep-23 7:02:54.024 [Endpoint/13]    [L0] core.endpoint.s3.ops.s3_put_object_uploadId:: PUT OBJECT PART bucket-11 big_file_100GB 988 bucket-1/big_file_100GB
/noobaa_init_files/noobaa_init.sh: line 75:    13 Killed                  $*




######################################################################
Thu Sep 23 07:02:57 UTC 2021 NooBaa: Process exited RIP (RC=137)
######################################################################

```

**Comment by akmithal** [comment]
[must-gather-2309.tar.gz](https://github.com/noobaa/noobaa-core/files/7215595/must-gather-2309.tar.gz)

**Comment by nimrod-becker** [comment]
@akmithal did we see this? should not be reproduced

**Comment by akmithal** [comment]
This issue seems to be fixed with ODF 4.9.4-7 build. Closing this defect.
